### PR TITLE
Feature/windows support

### PR DESCRIPTION
**Pull request by paulrobello**
feat: Added windows support for shell tab
fix: Exception when new version cant be fetched
fix: Exception when container env var does not have a value
fix: Bad env var parsing when value has an equal sign in it

### CODE REVIEW

**Comment by pommee** [comment]
Very nice, will do some testing.  

In the meantime, could you please edit the commit messages to be semantic?  
Otherwise I can fix this once it's time to merge.

**Comment by paulrobello** [comment]
I tested on Windows gitbash, Windows WSL Ubuntu, Mac OSX silicon.

**Comment by pommee** [comment]
Can see now that the branch is not available. Reason is propably due to me force-pushing while working with actions (apologize about that).  

Any chance you can re-create the branch from your fork?

**Comment by paulrobello** [comment]
Not sure what you mean.
The branch exists on my fork and wont exist on your repo until the merge request is accepted.

**Comment by paulrobello** [comment]
I resolved the conflicts

**Comment by paulrobello** [comment]
I hope resolving the conflicts in the PR allows it to show up properly for you.
Do you have everything you need from me?

File: application/main.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import click
 import yaml
 from colorama import Fore, Style
-from packaging.version import parse
+from packaging.version import parse, InvalidVersion
```
Comment (pommee) [inline application/main.py:9]: `InvalidVersion` not used. Can be removed.

File: application/widget/shell.py [hunk 2]
```diff
@@ -1,8 +1,13 @@
 import os
-import pty
+
```
Comment (pommee) [inline application/widget/shell.py:2]: Some formatting errors in this file.  

You can run `make format` to see what is causing this.  
Sometimes `make format-fix` can fix it for you (depending on the issue).

File: application/widget/shell.py [hunk 3]
```diff
@@ -44,22 +49,36 @@ def run_shell(self) -> None:
         self.container = self.docker_manager.selected_container
         self.output_widget = self.query_one("#shell-output", RichLog)
         self.input_widget = self.query_one("#shell-input", Input)
+        self.shell = "bash"
```
Comment (pommee) [inline application/widget/shell.py:52]: `default_shell = "bash"` or just remove this and set where it is used to "bash".  
I don't have a strong opinion.

File: pyproject.toml [hunk 4]
```diff
@@ -19,6 +19,8 @@ click = "^8.1.7"
 colorama = "^0.4.6"
 yaspin = "^3.0.2"
 packaging = "^24.1"
+cachetools = "^5.3.3"
```
Comment (pommee) [inline pyproject.toml:22]: `cachetools` is not used, can be removed.

**Review by pommee — COMMENTED** [review]
Smaller changes needed before we can merge :rocket:   

Please, also change the commit messages to be semantic.  

And the commit:
* "Merge branch 'main' into feature/windows_support"  

Can be squashed with your other commit related to poetry.lock.

**Comment by paulrobello** [comment]
Requested changes have been made

**Comment by pommee** [comment]
Nice work and thanks for the Windows fixes :rocket:

**Pull request merged**